### PR TITLE
Added install guide for nvm and rust to dev setup

### PR DESCRIPTION
**Pull request by NurramoX**
## Basics

- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

**Review by badnames — CHANGES_REQUESTED** [review]
Great guide, well done. 

Please provide a link to this document in the respective README files in frontend and backend, so that new developers can easily find this information.

**Review by aidnurs — APPROVED** [review]

File: doc/development_setup.md [hunk 1]
```diff
@@ -26,6 +26,43 @@ Rust formatting can be achieved by adding the following to settings.json in VSCo
...
+nvm use --lts
+```
+
+Or if you use the default macOS starting with Catalina shell `zsh`, try:
+
+```zsh
+sh -c "$(curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh)"
+```
+
+### Installing rust + cargo
```
Comment (markus2330) [inline doc/development_setup.md:45]: There are several capitalization problems, please make sure names and titles are capitalized, e.g.:

```suggestion
### Installing Rust + Cargo
```

File: doc/development_setup.md [hunk 2]
```diff
@@ -26,6 +26,43 @@ Rust formatting can be achieved by adding the following to settings.json in VSCo
...
+
+If you’re using Linux or macOS, open a terminal and enter the following command:
+
+```bash
+$ curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh
```
Comment (markus2330) [inline doc/development_setup.md:50]: This maybe does now the right thing, but it won't if Rust gets upgraded in future. In backend README we defined we want to use Rust 1.67.1, so please install this version.

File: doc/development_setup.md [hunk 3]
```diff
@@ -26,6 +26,43 @@ Rust formatting can be achieved by adding the following to settings.json in VSCo
...
+```
+This installs rust and cargo automatically on your machine. 
+If you get linker errors, proceed by installing a C Compiler, which will typically include a linker. 
+
+On Windows, go to https://www.rust-lang.org/tools/install and follow the instructions for installing Rust. At some point in the installation, you’ll receive a message explaining that you’ll also need the MSVC build tools for Visual Studio 2013 or later.
```
Comment (markus2330) [inline doc/development_setup.md:55]: Please read documentation guidelines (doc/documentation.md), we use line breaks for every sentence:
```suggestion
On Windows, go to https://www.rust-lang.org/tools/install and follow the instructions for installing Rust.
At some point in the installation, you’ll receive a message explaining that you’ll also need the MSVC build tools for Visual Studio 2013 or later.
```

**Review by markus2330 — APPROVED** [review]
Great job! Only some minor details. I did not try the tutorial, though.

**Comment by markus2330** [comment]
> Please provide a link to this document in the respective README files in frontend and backend, so that new developers can easily find this information.

I agree, good catch! Can you please also check for my requested changes when you review again? (@NurramoX request his review when you did the requested changes.)

**Comment by badnames** [comment]
> Can you please also check for my requested changes when you review again?

Yes of course.

**Review by badnames — APPROVED** [review]

**Comment by markus2330** [comment]
Great job!

**Pull request merged**